internal/api/v1/admin/webhooks/webhooks_api.go

```diff
@@ -45,6 +46,26 @@ func (a *API) UnsubscribeWebhook(ctx context.Context, cmd *commands.CancelWebhoo
 	return nil
 }
 
+func (a *API) AdminGetAllWebhooks(ctx context.Context) ([]*notifications.Webhook, error) {
+	var webhooks []*notifications.Webhook
+
+	resp, err := a.httpClient.
+		R().
+		SetContext(ctx).
+		SetResult(&webhooks).
+		Get(a.url.String())
+
+	if err != nil {
+		return nil, fmt.Errorf("HTTP request failure: %w", err)
+	}
+
+	if resp.IsError() {
+		return nil, fmt.Errorf("API error: %s", resp.String())
+	}
```

Comment:
The error is handled automatically, so you are not supposed to check `IsError` manually.
See how it's configured: https://github.com/bitcoin-sv/spv-wallet-go-client/blob/1eb35b59201aea82b7092208a2eca698cd7f844e/internal/restyutil/restyutil.go#L26-L34

```suggestion
	_, err := a.httpClient.
		R().
		SetContext(ctx).
		SetResult(&webhooks).
		Get(a.url.String())

	if err != nil {
		return nil, fmt.Errorf("HTTP request failure: %w", err)
	}
```

Also, as I can see in the codebase, request message are "almost always": `"HTTP response failure"` not `"request" failure`. I don't know why, but it's a thing 🤷  - @mgosek-4chain?